server/components/populate_sport_test.js

```diff
@@ -3,37 +3,40 @@
 **/
 
 var Converter = require('csvtojson').core.Converter;
-var db = require('../server/api/models');
+var db = require('../api/models');
 var Sport = db.Sport;
 var Player = db.Player;
 var fs = require('fs')
+var BPromise = require("sequelize/node_modules/bluebird");
 
 var Populate = (function(){
   var _parseCsv = function(csv, sport) {
     var fileStream = fs.createReadStream(csv);
     var csvConverter = new Converter({constructResult:true});
-    csvConverter.on("end_parsed",function(jsonObj){
-      populateDatabase(jsonObj, sport);
+    return BPromise.try(function() {
```

Comment:
There's a way to take arbitrary functions and make them asynchronous with
promises. Take a look at promisification:
https://github.com/petkaantonov/bluebird/blob/master/API.md#promisification

There's an example in our code in message.helper.js in the sendEmail
function where I make e-mail sending return a promise so we can wait on
that.

On Tue, Feb 10, 2015 at 9:50 AM, Sammy Rachman <notifications@github.com>
wrote:

> In server/components/populate_sport_test.js
> <https://github.com/ShaiNirShado/Shado/pull/6#discussion_r24416591>:
>
> >
> >  var Populate = (function(){
> >    var _parseCsv = function(csv, sport) {
> >      var fileStream = fs.createReadStream(csv);
> >      var csvConverter = new Converter({constructResult:true});
> > -    csvConverter.on("end_parsed",function(jsonObj){
> > -      populateDatabase(jsonObj, sport);
> > +    return BPromise.try(function() {
>
> I'm trying to fix the promise chain in this spec, which I think always
> gets broken when the CSV parser is run. I want to be able to actually have
> the promise chain incorporate the csv parser and wait until it finishes
> running so the spec works properly (in the sense that the chain isn't
> broken up at all so the first spec doesn't run until this is finished.)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ShaiNirShado/Shado/pull/6/files#r24416591>.
>